### DISCUSSION TITLE
CPU and memory requirement for etcd/Mino/Pulsar

### DISCUSSION THREAD

**Discussion by dsp05** [milvus-io/milvus#22574] [Q&A and General discussion]
Hi team, I'm experimenting with Milvus and trying to understand what the minimum requirement is, for e.g. 100m/256dim vectors.

From the size tool using IVF_SQ8, etcd requires 3 instances of 4core/8gb, Minio requires 4 instances of 4core/16gb, Plusar requires roughly a total of 14core/80gb, I wonder what the impact is if the allocated resource is lower than this limit. e.g.
- less CPU and memory for etcd/Plusar, or
- less than 4 Minio instances and less CPU and memory for each instance.

Thanks!

**Comment by xiaofan-luan** [comment]
> Hi team, I'm experimenting with Milvus and trying to understand what the minimum requirement is, for e.g. 100m/256dim vectors.
> 
> From the size tool using IVF_SQ8, etcd requires 3 instances of 4core/8gb, Minio requires 4 instances of 4core/16gb, Plusar requires roughly a total of 14core/80gb, I wonder what the impact is if the allocated resource is lower than this limit. e.g.
> 
> * less CPU and memory for etcd/Plusar, or
> * less than 4 Minio instances and less CPU and memory for each instance.
> 
> Thanks!

1. if you deploy milvus on cloud, then directly use cloud offering would be highly recommended.
2. you can try to shrink all the services to half, everything should work fine if you don't insert/delete frequently I guess.
3. With smaller minio and pulsar, recovery speed might be a little bit slower.
4. I'm totally fine with 2c/4G etcd config

**Reply by dsp05** [reply]
Thanks! What is the minimum number of Mino instances? Will fewer instances e.g. only two cause any issues or it just slows down recovery?

**Reply by xiaofan-luan** [reply]
> Thanks! What is the minimum number of Mino instances? Will fewer instances e.g. only two cause any issues or it just slows down recovery?

I thought it has to be at least 3 to keep your data safe.  @LoveEachDay any ideas?
